pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -1585,11 +1585,128 @@ func (s *PluginSuite) TestCreateRegistrationEntry() {
 		registrationEntry, err := s.ds.CreateRegistrationEntry(ctx, validRegistrationEntry)
 		s.Require().NoError(err)
 		s.Require().NotNil(registrationEntry)
-		s.NotEmpty(registrationEntry.EntryId)
-		registrationEntry.EntryId = ""
-		s.assertCreatedAtField(registrationEntry, now)
-		registrationEntry.CreatedAt = validRegistrationEntry.CreatedAt
-		s.RequireProtoEqual(registrationEntry, validRegistrationEntry)
+		s.assertEntryEqual(s.T(), validRegistrationEntry, registrationEntry, now)
+	}
+}
+
+func (s *PluginSuite) TestCreateOrReturnRegistrationEntry() {
+	now := time.Now().Unix()
+
+	for _, tt := range []struct {
+		name          string
+		updateEntry   func(*common.RegistrationEntry) *common.RegistrationEntry
```

Comment:
```suggestion
		cloneAndModifyEntry   func(*common.RegistrationEntry) *common.RegistrationEntry
```
or similar name?
`updateEntry`, especially given Server and DB API names, has different connotations

Reply:
I'll update to modifyEntry, this function is not cloning entry